docs/rfq-and-price-oracle.md

```diff
@@ -0,0 +1,233 @@
+amount requirements), overpaying by `0.4211` cents.
+
+If the user's wallet decided to split up the payment into 16 parts for example,
+then each part would correspond to `3289.4737` cents. To satisfy the integer
+asset amount and invoice minimum amount requirement, each of the 16 HTLCs would
+send out `3290` cents. That's a full `8.4211` cents of overpayment.
+
+## What precision should I choose when minting an asset for the Lightning Network?
+
+To address the issue of rounding up when splitting payments or representing
+small satoshi amounts as asset units, an issuer of assets should use a high
+enough value for `decimal_display` when minting.
+
+**But what is a good value for `decimal_display`?**
+
+We recommend to use a `decimal_display` value of `6` for currencies which
+use a smaller subunit with two decimal places (such as cents for USD or EUR,
+penny for GBP and so on).
+
+For currencies without smaller units (for example JPY or VND), a
+`decimal_display` value of `4` is recommended.
+
+# RFQ
+
+The RFQ system is responsible for acquiring real-time price quotes for
+converting between asset units and satoshi (both directions) or between
+different asset types.
+
+There are two main user stories (as seen from the point of view of the wallet
+end user):
+
+1. Sending out assets: The user wants to pay a Lightning Network invoice that
+   is denominated in satoshi. The user only has assets in their wallet, they
+   (or their wallet software) want to find out how many asset units they need to
+   send in order to satisfy the invoice amount in satoshi.
+2. Receiving assets: The user wants to get paid in a specific asset. The user
+   only knows about the asset, so they (or their wallet software) want to find
+   out what the asset amount corresponds to in satoshi.
+
+## Sell order
+
+The sell order covers the first user story: The user wants to pay a
+satoshi-denominated invoice with assets. The end result is that the user sells
+some of their assets in their channel to the RFQ peer (edge node), requesting
+them to forward satoshi to the network.
+
+Formal definition (TODO: refine):
+- Use case: sending assets as a payment, selling `buxx` for `msat`
+- User query: `Q = how many out_asset units for in_asset amount?` (how many
+  `buxx` do I need to seel to pay this payment denominated in `msat`?)
+- `out_asset`: `buxx` (user sells `buxx` asset to RFQ peer)
+- `in_asset`: `msat` (user "receives" `msat` from RFQ peer, which are then
+  routed to the network)
+- `max_amount`: `in_asset` (what is the maximum amount of `msat` the RFQ peer
+  has to forward to the network? Equal to invoice amount plus user-defined max
+  routing fee limit)
+- `price_out_asset`: `out_asset_units_per_btc` (`buxx per BTC`)
+- `price_in_asset`: `in_asset_units_per_btc` (`msat per BTC`)
+
+## Buy order
+
+The buy order covers the second user story: The user wants to get paid, they
+create a satoshi-denominated invoice from their chosen asset amount they want to
```

Comment:
I think instead we should state that they create an invoice in terms of _asset units_, which is then mapped to a normal satoshi-denominated invoices.